File: frb_example/pure_dart/lib/src/rust/api/uuid_type.dart
// ignore_for_file: invalid_use_of_internal_member, unused_import

import 'package:flutter_rust_bridge/flutter_rust_bridge_for_generated.dart';

import '../frb_generated.dart';

Future<UuidValue> handleUuid({required UuidValue id, dynamic hint}) =>
    RustLib.instance.api.handleUuid(id: id, hint: hint);

Future<List<UuidValue>> handleUuids({required List<UuidValue> ids, dynamic hint}) =>
    RustLib.instance.api.handleUuids(ids: ids, hint: hint);

Future<FeatureUuid> handleNestedUuids({required FeatureUuid ids, dynamic hint}) =>
    RustLib.instance.api.handleNestedUuids(ids: ids, hint: hint);

class FeatureUuid {
  final UuidValue one;
  final List<UuidValue> many;

  const FeatureUuid({
    required this.one,
    required this.many,
  });

  @override
  int get hashCode => one.hashCode ^ many.hashCode;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is FeatureUuid && runtimeType == other.runtimeType && one == other.one && many == other.many;
}

File: frb_example/old/pure_dart/rust/src/temp/uuid_type.rs
#[derive(Debug, Clone)]
pub struct FeatureUuid {
    pub one: uuid::Uuid,
    pub many: Vec<uuid::Uuid>,
}

pub fn handle_uuid(id: uuid::Uuid) -> anyhow::Result<uuid::Uuid> {
    Ok(id)
}

pub fn handle_uuids(ids: Vec<uuid::Uuid>) -> anyhow::Result<Vec<uuid::Uuid>> {
    Ok(ids)
}

pub fn handle_nested_uuids(ids: FeatureUuid) -> anyhow::Result<FeatureUuid> {
    Ok(ids)
}
